src/modules/esl-utils/dom/rect.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * A Rect describes the size and position of a rectangle.
+ * */
+export class Rect {
+  /**
+   * The X coordinate of the Rect's origin (top-left corner of the rectangle).
+   * */
+  public x = 0;
+  /**
+   * The Y coordinate of the Rect's origin (top-left corner of the rectangle).
+   * */
+  public y = 0;
+  /**
+   * The width of the Rect.
+   * */
+  public width = 0;
+  /**
+   * The height of the Rect.
+   * */
+  public height = 0;
+
+  /**
+   * The static method creates a new Rect instance from a rect-like object.
+   * @param rect - rect-like object
+   * */
+  public static from(rect: {x?: number; y?: number; width?: number; height?: number} = {}): Rect {
+    return new this(
+      rect.x ?? 0,
+      rect.y ?? 0,
+      rect.width ?? 0,
+      rect.height ?? 0,
+    );
+  }
+
+  public constructor(x?: number, y?: number, width?: number, height?: number) {
+    if (x != null) {
```

Comment:
It looks like if we make it undefined or null it will be 0.
So propose to make 
```
this.x = x || 0;
this.y = y || 0;
this.width= width || 0;
this.height = height || 0;
```
in the constructor and then get rid of properties defaults (as they will add additional transpiled code)
and get rid of all checks in method `from` above 
`return new this(rect.x, rect.y, rect.width, rect.height); `

Reply:
Fixed